src/gevent_server.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from gevent.server import StreamServer
+from gevent.monkey import patch_all
+import sys
+from server import parse_request, server_read, response_ok, server_response, response_error
+
+IP = '127.0.0.1'
+PORT = int(sys.argv[1])
+
+def server_handler(connection, address):
+    """Handles the gevent StreamServer for individual clients."""
+    try:
+        result, mime = parse_request(server_read(connection))
+        print("log:", result)
+        to_send = response_ok(result, mime)
+        server_response(to_send, connection)
+    except Exception as error:
+        try:
+            error = error.args[0]
+            code = int(error.split(':')[0])
+            error = error.split(':')[1].strip()
```

Comment:
notice that you call `error.split` twice, just to get at the first and second elements in the tuple it's going to return?  How about this instead:

```python
code, error = error.split(':')
error = error.strip()
```

I'm betting that you don't actually need to convert the code into an int, unless you are using that to separate errors that are supposed to make HTTP responses from ones that should become 500 errors.